flang/include/flang/Lower/PFTBuilder.h

```diff
@@ -247,6 +247,25 @@ struct Evaluation : EvaluationVariant {
     }});
   }
 
+  EvaluationList &getNestedEvaluations() {
+    assert(evaluationList && "no nested evaluations");
+    return *evaluationList;
+  }
+
+  bool hasAtLeastOneNestedEvaluation() const {
+    return evaluationList && !evaluationList->empty();
+  }
+
+  Evaluation &getFirstNestedEvaluation() {
+    assert(hasAtLeastOneNestedEvaluation() && "no nested evaluations");
```

Comment:
Should we consider forcing this condition to always be tested?

```
if (hasAtLeastOneNestedEvaluation()) return evalList->front();
llvm_unreachable("no nested evals");
```

Reply:
I see your point, that goes along the discussion from last Wednesday that LLVM disable asserts in production... I think most asserts in the bridge are cheap and should also run in production though and I find it it a bit more readable. So I would rather keep it like an assert and ensure that all the asserts in our code are on if we want to.